### ISSUE TITLE
Building gbdk-lib on ubuntu fails

### ISSUE THREAD

**Issue by arikami** [Zal0/gbdk-2020#28]
```
make -C libc
make[1]: Entering directory '/mnt/c/development/magazine_repo/gbdk-2020/gbdk-lib/libc'
for i in gbz80; do make port THIS=$i PORT=$i; done
make[2]: Entering directory '/mnt/c/development/magazine_repo/gbdk-2020/gbdk-lib/libc'
rm -f  *.o *.cdb *.sym *.lst *~ *.asm
if [ -e global.s ]; then \
        sed -e "s/.NEAR_CALLS\W=\W[0-9]\+/.NEAR_CALLS = 1/" global.s > tmp1.txt ;\
        mv tmp1.txt global.s; \
fi
mkdir -p ../build/small/asxxxx/gbz80
/bin/sdcc -I../include -DSDCC_z80 -D__gbz80 -mgbz80 -DGBDK=1 -c -o ../build/small/asxxxx/gbz80/bsearch.o bsearch.c
make[2]: /bin/sdcc: Command not found
../libc/rules-asxxxx.mk:7: recipe for target '../build/small/asxxxx/gbz80/bsearch.o' failed
make[2]: *** [../build/small/asxxxx/gbz80/bsearch.o] Error 127
make[2]: Leaving directory '/mnt/c/development/magazine_repo/gbdk-2020/gbdk-lib/libc'
Makefile:31: recipe for target 'ports' failed
make[1]: *** [ports] Error 2
make[1]: Leaving directory '/mnt/c/development/magazine_repo/gbdk-2020/gbdk-lib/libc'
Makefile:4: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by basxto** [comment]
It should say something like this: 
```
$ /bin/sdcc -v
SDCC : mcs51/z80/z180/r2k/r3ka/gbz80/tlcs90/ez80_z80/ds390/pic16/pic14/TININative/ds400/hc08/s08/stm8/pdk13/pdk14/pdk15 4.0.0 #11528 (Linux)
published under GNU General Public License (GPL)
```
Otherwise that’s the wrong directory … `which sdcc`

**Comment by DaSalba** [comment]
> which sdcc | sed "s/\(.*\)bin\/sdcc/SDCCDIR=\1/g"

That is a good one, perhaps GBDK should use it instead of having the user define the env variable...

**Comment by basxto** [comment]
I’ve  built a makefile for building a .deb file on debian/ubuntu, it’s neither really tested nor maintained. I use neither of them.
```sh
git clone https://github.com/basxto/gbdk-2020
cd gbdk-2020
git checkout deb-package
make -C deb-package
dpkg -i deb-package/gbdk-2020.deb
```
This additionally needs `dpkg-deb` installed.

**Comment by nazebzurati** [comment]
Hi, I also got the same issue and I manage to fixed that. Here is what I've done:

1. Download sdcc from [here](http://sdcc.sourceforge.net/snap.php) and extract - the latest 'Linux on AMD64' snapshot link is broken and I have to use the 24th June snapshot.
2. Export the path snapshot path using command:-
   ```bash
   export SDCCDIR=path/to/sdcc/snapshot
   ```
3. Compile GBDK-2020

Eventually, you'll get another error which as shown below:
```bash
cp: cannot stat '<some/path>/bin/{as2gbmap,makebin,packihx,sdar,sdasgb,sdcc,sdcdb,sdcdb{,src}.el,sdcpp,sdldgb,sdnm,sdobjcopy,sdranlib,sz80}': No such file or directory
make: *** [Makefile:213: sdcc-install] Error 1
```

To fix it, you can go to line 216 in ./Makefile and update the line.
```
# before
@cp $(SDCCDIR)/bin/{as2gbmap,makebin,packihx,sdar,sdasgb,sdcc,sdcdb,sdcdb{,src}.el,sdcpp,sdldgb,sdnm,sdobjcopy,sdranlib,sz80} $(BUILDDIR)/bin/SDCC/bin/
# after
@cp $(SDCCDIR)/bin/* $(BUILDDIR)/bin/SDCC/bin/
```

.. and run make command again. Voila!

I'm using Xubuntu 20.04 and tested on [GBDK-2020 v3.2](https://github.com/Zal0/gbdk-2020/releases/tag/v3.2).

**Comment by JL2210** [comment]
That `@cp $(SDCCDIR)/bin/* $(BUILDDIR)/bin/SDCC/bin/` just brings back the problem we had before: if you have SDCC in /usr it copies *all* the binaries from /usr/bin to the GBDK installation directory.

**Comment by nazebzurati** [comment]
Yes, I understand.

The fix only to covers the globbing issue if we use standalone sdcc binaries, instead of the one that installed in the system.

I'll try to support _SDCCDIR=/usr_ case as well in the same fix.

**Comment by untoxa** [comment]
fixed in 4.0 release
